src/Model/Request.php

```diff
@@ -17,13 +17,16 @@ class Request
     private $data;
     /** @var string */
     private $requestId;
+    /** @var string */
+    private $responseClass;
 
-    public function __construct(string $path, string $method, array $data = [], string $requestId = null)
+    public function __construct(string $path, string $method, array $data = [], string $requestId = null, string $responseClass = null)
```

Comment:
I don't like idea that request knows about response. This should be responsibility of `ResponseDecoder`.

Reply:
+1. Also ResponseDecoder gets only the `PSR ResponseInterface` as parameter, so we'd have to put the pairing between requests & responses into `RequestManager`.

To me it feels better to pair a response directly with request.